ipfshttpclient/client/__init__.py

```diff
@@ -254,19 +260,55 @@ def add_bytes(self, data: bytes, **kwargs):
 		----------
 		data
 			Content to be added as a file
+		trickle
+			Use trickle-dag format (optimized for streaming) when generating
+			the dag; see `the old FAQ <https://github.com/ipfs/faq/issues/218>`_
+			for more information
+		only_hash
+			Only chunk and hash, but do not write to disk
+		chunker
+			The chunking algorithm to use
+		pin
+			Pin this object when adding
+		raw_leaves
+			Use raw blocks for leaf nodes. (experimental). (Default: ``True``
+			when *nocopy* is True, or ``False`` otherwise)
+		cid_version
+			CID version. Default value is provided by IPFS daemon. (experimental)
 
 		Returns
 		-------
 			str
 				Hash of the added IPFS object
 		"""
+
+		opts = {
+			"trickle": trickle,
+			"only-hash": only_hash,
+			"pin": pin,
+			"raw-leaves": raw_leaves if raw_leaves is not None else nocopy,
+		}  # type: ty.Dict[str, ty.Union[str, bool]]
+		for option_name, option_value in [
+			("chunker", chunker),
+			("cid-version", cid_version),
+		]:
+			if option_value is not None:
+				opts[option_name] = option_value
+		kwargs.setdefault("opts", {}).update(opts)
+
 		body, headers = multipart.stream_bytes(data, chunk_size=self.chunk_size)
 		return self._client.request('/add', decoder='json',
 		                            data=body, headers=headers, **kwargs)
 
 	@utils.return_field('Hash')
 	@base.returns_single_item(dict)
-	def add_str(self, string, **kwargs):
+	def add_str(self, string,
+	        trickle: bool = False, only_hash: bool = False,
+	        chunker: ty.Optional[str] = None,
+	        pin: bool = True, raw_leaves: bool = None,
+	        cid_version: ty.Optional[int] = None,
+	        **kwargs: base.CommonArgs):
```

Comment:
Same issues as with add_bytes - parameter on each line, and add return type declaration.

Reply:
Done.